audio-worklet/src/noisegate-script-processor.js

```diff
@@ -0,0 +1,187 @@
+
+    // The noise gate is connected to and from by dummy input and output nodes.
+    this.input = new GainNode(this.context_);
+    this.output = new GainNode(this.context_);
+    this.input.connect(this.node_).connect(this.output);
+
+    // The last envelope level of a given buffer.
+    this.lastLevel_ = 0;
+
+    // The last weight (between 0 and 1) assigned, where 1 means the gate
+    // is open and 0 means it is closed and the sample in the output buffer is
+    // muted.
+    this.lastWeight_ = 1.0;
+    this.channel_ = new Float32Array(this.node_.bufferSize);
+    this.envelope_ = new Float32Array(this.node_.bufferSize);
+    this.weights_ = new Float32Array(this.node_.bufferSize);
+  }
+
+  /**
+   * Gradually mutes input which registers beneath specified threshold.
+   * @param {AudioProcessingEvent} event event object containing
+   *                                     input and output buffers
+   */
+  onaudioprocess_(event) {
+    // Stereo input is downmixed to mono input via averaging.
+    if (event.inputBuffer.numberOfChannels === 2) {
+      for (let i = 0; i < event.inputBuffer.length; i++) {
+        this.channel_[i] = (event.inputBuffer.getChannelData(0)[i] +
+                           event.inputBuffer.getChannelData(1)[i]) / 2;
+      }
+    } else {
+      this.channel_ = event.inputBuffer.getChannelData(0);
+    }
+
+    let envelope = this.detectLevel_(this.channel_);
+
+    for (let i = 0; i < event.inputBuffer.numberOfChannels; i++) {
+      let input = event.inputBuffer.getChannelData(i);
+      let output = event.outputBuffer.getChannelData(i);
+      let weights = this.computeGain_(envelope);
+
+      for (let j = 0; j < input.length; j++) {
+        output[j] = weights[j] * input[j];
+      }
+    }
+  }
+
+  /**
+   * Detect level using the difference equation for the Root Mean Squared
+   * value of the input signal.
+   * @param {Float32Array} input audio buffer
+   * @return {Float32Array} the level of the signal
+   */
+  detectLevel_(channel) {
+    // The signal level is determined by filtering high frequency oscillation
+    // with exponential smoothing.
+    // This is equivalent to computing the root-mean-square (RMS) value
+    // of the signal. See http://www.aes.org/e-lib/browse.cfm?elib=16354 for
+    // details.
+    this.envelope_[0] = this.alpha_ * this.lastLevel_ +
+        (1 - this.alpha_) * Math.pow(channel[0], 2);
+
+    for (let j = 1; j < channel.length; j++) {
+      this.envelope_[j] = this.alpha_ * this.envelope_[j - 1] +
+          (1 - this.alpha_) * Math.pow(channel[j], 2);
+    }
+    this.lastLevel_ = this.envelope_[this.envelope_.length - 1];
+
+    // For sine waves, the envelope eventually reaches an average power of
+    // a^2 / 2. Sine waves are therefore scaled back to the original amplitude,
+    // but other waveforms or constant sources can only be approximated.
+    for (let k = 0; k < this.envelope_.length; k++) {
```

Comment:
I think this is wrong.  Your filter should be exactly
```
y(n) = alpha*y(n-1)+(1-alpha)*x(n)^2
```
After computing this, you scale `y(n)` by sqrt(2).  And this scaling gets propagated to the next audioprocess event.

For simplicity, you should separate out this scaling so that the filter values aren't scaled.